manifests/init.pp

```diff
@@ -49,47 +58,60 @@
   }
 
   podman::secret { $server_cert_secret_name:
-    path => $certs::iop_advisor_engine::server_cert,
+    ensure => $ensure,
+    path   => $certs::iop_advisor_engine::server_cert,
   }
 
   podman::secret { $server_key_secret_name:
-    path => $certs::iop_advisor_engine::server_key,
+    ensure => $ensure,
+    path   => $certs::iop_advisor_engine::server_key,
   }
 
   podman::secret { $server_ca_cert_secret_name:
-    path => $certs::iop_advisor_engine::server_ca_cert,
+    ensure => $ensure,
+    path   => $certs::iop_advisor_engine::server_ca_cert,
   }
 
   podman::secret { $client_cert_secret_name:
-    path => $certs::iop_advisor_engine::client_cert,
+    ensure => $ensure,
+    path   => $certs::iop_advisor_engine::client_cert,
   }
 
   podman::secret { $client_key_secret_name:
-    path => $certs::iop_advisor_engine::client_key,
+    ensure => $ensure,
+    path   => $certs::iop_advisor_engine::client_key,
   }
 
   podman::secret { $client_ca_cert_secret_name:
-    path => $certs::iop_advisor_engine::client_ca_cert,
+    ensure => $ensure,
+    path   => $certs::iop_advisor_engine::client_ca_cert,
+  }
+
+  $service_ensure = $ensure ? {
+    'present' => true,
+    'absent'  => false,
   }
 
   file { $log_dir:
-    ensure => directory,
+    ensure => $directory_ensure,
+    force  => true,
     mode   => '0755',
     owner  => 'root',
     group  => 'root',
   }
 
   ['uploads', 'failed', 'logs'].each |String $file| {
     file { "${log_dir}/${file}":
```

Comment:
I just realized that placing "uploads" and "failed" under log_dir is kinda wrong?